### PR TITLE
Add file comments from email replies

### PR DESCRIPTION
**Pull request by adam-thomas**
Users can make file comments on discussions by attaching files to their email replies.  At the moment, this just generates a (separate) `FileComment` per attachment on top of the normal `TextComment` added by the email body.  The `TextComment` comes last as per @meshy's suggestion, but I'm happy to reorder them.

File comments don't currently display properly - they really ought to link to the uploaded file and say "Uploaded a file: [filename]" rather than just "/groups/comments/[filename]" with no linking.  I'll fix that next.

I've tested this on the server and it worked!  First try!  This is entirely due to me finally finding some Mailgun API description... in a [blog post from 2011](http://blog.mailgun.com/handle-incoming-emails-like-a-pro-mailgun-api-2-0/).  Sigh.

@incuna/backend review please? :)

### CODE REVIEW

File: groups/views/comments.py [hunk 1]
```diff
@@ -125,17 +125,38 @@ def extract_uuid_from_email(email, request):
...
+        Mailgun provides an entry called `attachment-count` to store the number
+        of attachments, then each attachment is a separate entry, `attachment-x` where
+        `x` is a number.
+        """
+        for attachment in request.FILES.values():
+            models.FileComment.objects.create(
+                file=attachment,
+                user=user,
+                discussion=discussion,
+            )
```
Comment (KevinEtienne) [inline groups/views/comments.py]: Do we need some way to filter down the file types?

Comment (adam-thomas) [inline groups/views/comments.py]: Not as far as I know.

**Comment by adam-thomas** [comment]
Any more comments?

**Comment by KevinEtienne** [comment]
Looks good

**Comment by adam-thomas** [comment]
Thanks :)

**Pull request merged**